frame/staking/src/pallet/impls.rs

```diff
@@ -1263,9 +1257,18 @@ where
 					}
 				} else {
 					// Defer to end of some `slash_defer_duration` from now.
-					<Self as Store>::UnappliedSlashes::mutate(active_era, move |for_later| {
-						for_later.push(unapplied)
-					});
+					log!(
+						debug,
+						"deferring slash of {:?}% happened in {:?} (reported in {:?}) to {:?}",
+						slash_fraction,
+						slash_era,
+						active_era,
+						slash_era + slash_defer_duration + 1,
+					);
+					<Self as Store>::UnappliedSlashes::mutate(
+						slash_era + slash_defer_duration + 1,
```

Comment:
should we care about safe math here?

Reply:
To the point that era numbers saturate, so both yes and no. I will make it saturating.